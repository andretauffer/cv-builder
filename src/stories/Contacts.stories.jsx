import React, { useState, useEffect } from 'react';

import content from "../content2.json";
import Contacts from "../components/ContactsCard";
import { Resolver } from "@stoplight/json-ref-resolver";


import { Context } from "../Context";
const resolver = new Resolver();

const ContactsSignature = (args, context) => {
  const [resolvedContent, setContent] = useState(null);
  useEffect(() => {
    resolver.resolve(content).then(resolved => setContent(resolved.result))
  }, []);

  return resolvedContent ? <Contacts {...{ ...resolvedContent.intro.contacts }} /> : <></>;
}


export default {
  title: 'Components/Contacts',
  component: ContactsSignature,
};

const Template = (args) => <ContactsSignature {...args} />;

export const AllContacts = Template.bind({});
AllContacts.decorators = [
  (Story) => {

    return <div>

      <Context.Provider
        value={{
          keywords: []
        }}>
        <Story />
      </Context.Provider>
    </div>
  }
];
