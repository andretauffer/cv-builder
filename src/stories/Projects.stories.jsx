import React, { useState, useEffect } from 'react';

import content from "../content2.json";
import Projects from "../components/Projects";
import { Resolver } from "@stoplight/json-ref-resolver";


import { Context } from "../Context";
const resolver = new Resolver();

const ProjectsSignature = (args, context) => {
  const [resolvedContent, setContent] = useState(null);
  useEffect(() => {
    resolver.resolve(content).then(resolved => setContent(resolved.result))
  }, []);

  return resolvedContent ? <Projects {...{ projects: resolvedContent.projects.projects }} /> : <></>;
}


export default {
  title: 'Components/Projects',
  component: ProjectsSignature,
};

const Template = (args) => <ProjectsSignature {...args} />;

export const AllProjects = Template.bind({});
AllProjects.decorators = [
  (Story) => {

    return <div>

      <Context.Provider
        value={{
          keywords: []
        }}>
        <Story />
      </Context.Provider>
    </div>
  }
];

export const Frontend = Template.bind({});
Frontend.decorators = [
  (Story) => {

    return <div>

      <Context.Provider
        value={{
          keywords: ["Frontend"]
        }}>
        <Story />
      </Context.Provider>
    </div>
  }
];

export const Backend = Template.bind({});
Backend.decorators = [
  (Story) => {

    return <div>

      <Context.Provider
        value={{
          keywords: ["Backend"]
        }}>
        <Story />
      </Context.Provider>
    </div>
  }
];

