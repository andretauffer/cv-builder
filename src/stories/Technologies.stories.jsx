import React, { useState, useEffect } from 'react';

import content from "../content2.json";
import Technologies from "../components/Technologies";
import { Resolver } from "@stoplight/json-ref-resolver";


import { Context } from "../Context";
const resolver = new Resolver();

const TechnologiesSignature = (args, context) => {
  const [resolvedContent, setContent] = useState(null);
  useEffect(() => {
    resolver.resolve(content).then(resolved => setContent(resolved.result))
  }, []);

  return resolvedContent ? <Technologies {...{ technologies: resolvedContent.technologies.technologies }} /> : <></>;
}


export default {
  title: 'Components/Technologies',
  component: TechnologiesSignature,
};

const Template = (args) => <TechnologiesSignature {...args} />;

export const AllTechs = Template.bind({});
AllTechs.decorators = [
  (Story) => {
    
    return <div>

      <Context.Provider
        value={{
          keywords: []
        }}>
        <Story  />
      </Context.Provider>
    </div>
  }
];

