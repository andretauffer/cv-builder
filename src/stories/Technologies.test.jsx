import React from 'react';
// import '@testing-library/jest-dom/extend-expect'
import * as stories from "./Technologies.stories.jsx";

import { render } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';
// import 'jest-styled-components'

const { AllTechs } = composeStories(stories);

describe("The Technologies view snapshots", () => {

  it("renders with all icons when no keywords are selected", () => {
    const { asFragment } = render(<AllTechs />);
    expect(asFragment()).toMatchSnapshot();
  });
});
