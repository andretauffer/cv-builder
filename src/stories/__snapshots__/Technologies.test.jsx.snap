// Vitest Snapshot v1

exports[`The Technologies view snapshots > renders with all icons when no keywords are selected 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
